### ISSUE TITLE
Mono: Building project solution doesn't detect that .cs scripts were deleted

### ISSUE THREAD

**Issue by MrMaidx** [godotengine/godot#12415]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Windows 10, VS2017 amd64 compiler, latest master 2987e6c

**Issue description:**
When I delete .cs scripts (even from the editor) it seems that .csproj file is not automaticaly edited and deleted scripts are still listed there. This results in Build error when trying to run such project because build system cannot find deleted scripts.

When .csproj file is edited by hand and lines with deleted files are removed it seems to run correctly.

I'm not sure if this is a bug or just not implemented feature but an issue for it hopefully won't hurt. :)

**Steps to reproduce:**
Create a .cs script in any project.
Delete said script.
Running the project will result in Build error.

**Comment by Zylann** [comment]
I thought this was by design? I asked to @neikeq on Discord a few days ago, and Godot doesnt overwrite the solution everytime a script is added or removed, which is good because it lets you configure the solution the way you want. What you describe happens to any C# project anyways, like if you delete a .cs file outside of the project, it obviously fails once you try to build.

**Comment by MrMaidx** [comment]
Oh, okay. But this happens even when I delete scripts from FileSystem tab (right click - delete). I was thinking that when deleting from inside of the project it could edit the .csproj so the project can run from the editor.

**Comment by Zylann** [comment]
@MrMaidx Ah sorry I was ambiguous: if you delete it from Godot it won't update the csproj, but it will update if you use an IDE to do it. That was my point in fact, because unlike GDScript, C# needs a .csproj file to build, which is usually edited from an IDE.

**Comment by neikeq** [comment]
If the file is deleted from the file system, then we should remove it from the project. It just wasn't done yet. There is also the problem that when you remove a C# file outside of Godot, then it becomes a built-in script in Godot, which shouldn't happen...

**Comment by 11clock** [comment]
I honestly think a lot of problems would be solved if the solution was just rebuilt before every play. Delete them and then generate them again.

**Comment by Zylann** [comment]
@11clock please no, that would be unnecessary half of the time. A more appropriate time is when a C# file is added, changed or modified (so only when reaaally needed). Unity does exactly that, even though I still dislike it a bit (project gets overwritten, csproj options get lost, and have to wait everytime for compilation process which can take a while on a large project). Maybe Godot can do it better...

**Comment by akien-mga** [comment]
Related to #12917 (same bug, different symptom I'd say).

**Comment by neikeq** [comment]
Waiting on #10946 in order to fix #12415 and #12917.

**Comment by reduz** [comment]
@neikeq what should we do with this?

**Comment by akien-mga** [comment]
Duplicate of #12917.
